python/lwt_interface/tskit_lwt_interface.h

```diff
@@ -1334,6 +1335,27 @@ parse_table_collection_dict(tsk_table_collection_t *tables, PyObject *tables_dic
         }
     }
 
+    /* time_units */
+    value = get_table_dict_value(tables_dict, "time_units", false);
+    if (value == NULL) {
+        goto out;
+    }
+    if (value != Py_None) {
+        if (!PyUnicode_Check(value)) {
+            PyErr_Format(PyExc_TypeError, "'time_units' is not a string");
+            goto out;
+        }
+        time_units = parse_unicode_arg(value, &time_units_length);
+        if (time_units == NULL) {
```

Comment:
We should try to cover this failure if we can - maybe pass in a non-utf8 encodeable string? Might not be that easy to do though, I guess.

Reply:
Yep extra test with `"\ud861\udd37"` added.